### DISCUSSION TITLE
How do I reset my view?

### DISCUSSION THREAD

**Discussion by Tommycore** [microsoft/vscode-discussions#142] [Extension Development QnA] [answered]
UPDATE (22.10.2022): It's not a feature. Yet. You can help by upvoting it. We need 20 upvotes in 60 days. See https://github.com/microsoft/vscode/issues/162366.

As an extension developer, I want to be able to reset my views as if I had just opened my extension for the first time. All views in their original panels, all views hidden or shown as per their default settings, and all in the size the defaults dictate. How do I do this? I tried calling `View: Reset Editor Group Sizes` and `View: Reset View Locations`, but my four different custom views in my custom panel still have differing sizes, though in my `package.json` they're all set to `size: 1`.

**Comment by alefragnani** [comment, marked as answer]
Hi @Tommycore ,

You could try using the `yourView.resetViewLocation` command, where `yourView` is the name you used in the `contributions` section in `package.json`.

For instance, your `package.json`:

```json
        "contributes": {
		"views": {
			"project-manager": [
				{
					"id": "projectsExplorerFavorites",
					"name": "Favorites"
				},
```

Then you use

```ts
    await vscode.commands.executeCommand("projectsExplorerFavorites.resetViewLocation");
```

I did a small test and it _kind of work_,

* It moves the views back it its original container
* It does not move to the _original_ location (order), if the container has more than one view (but I didn't reset all, so maybe that's the reason)
 
I guess it is a good start for you.

Hope this helps

**Reply by Tommycore** [reply]
Hi @alefragnani,

thanks for the quick answer. Unfortunately that is not quite what I was looking for, as it only partly resets the views to their original form. I wrote a little script that loops through all my views and calls the command on every single one, but it had the same effect as using the `View: Reset View Locations` command from the command palette. It didn't reset the correct size (or order).

**Reply by alefragnani** [reply]
Yep... as I said, it could be a start, because even on a basic test, it didn't restored the original order.

I don't use the `size` attribute, and if I remember correctly, it is fairly new. So I wonder, if this is not a _bug_ in the `resetViewLocation` command. Maybe the attribute is not being reloaded in the reset process. 

If that's the case, and no other API/command is reported here, I suggest you to open an issue in the VS Code repo, so the team can take a look and fix it, if necessary.

Hope this helps

**Reply by Tommycore** [reply]
It does. I'll wait another 24h to see if anybody else can say something about it, and if nothing new comes up, I'll file a bug report and mark your post as answer. Thanks =)

**Reply by Tommycore** [reply]
As promised: https://github.com/microsoft/vscode/issues/162366
